typescript/sdk/src/token/deploy.ts

```diff
@@ -135,50 +137,63 @@ export class HypERC20Deployer extends GasRouterDeployer<
     chain: ChainName,
     config: HypERC20CollateralConfig,
   ): Promise<HypERC20Collateral> {
-    let contractName:
+    let tokenType:
       | TokenType.fastCollateral
       | TokenType.collateral
       | TokenType.collateralVault;
     switch (config.type) {
       case TokenType.fastSynthetic || TokenType.fastCollateral:
-        contractName = TokenType.fastCollateral;
+        tokenType = TokenType.fastCollateral;
         break;
       case TokenType.collateral:
-        contractName = TokenType.collateral;
+        tokenType = TokenType.collateral;
         break;
       case TokenType.collateralVault:
-        contractName = TokenType.collateralVault;
+        tokenType = TokenType.collateralVault;
         break;
       default:
         throw new Error(`Unknown collateral type ${config.type}`);
     }
-    return this.deployContract(chain, contractName, [
-      config.token,
-      config.mailbox,
-    ]);
+    return this.deployContractWithName(
+      chain,
+      tokenType,
+      hypERC20contracts[tokenType],
+      [config.token, config.mailbox],
+    );
   }
 
   protected async deployNative(
     chain: ChainName,
     config: HypNativeConfig,
   ): Promise<HypNative> {
     if (config.scale) {
-      return this.deployContract(chain, TokenType.nativeScaled, [
-        config.scale,
-        config.mailbox,
-      ]);
+      return this.deployContractWithName(
+        chain,
+        TokenType.nativeScaled,
+        hypERC20contracts[TokenType.nativeScaled],
+        [config.scale, config.mailbox],
+      );
     } else {
-      return this.deployContract(chain, TokenType.native, [config.mailbox]);
+      return this.deployContractWithName(
+        chain,
+        TokenType.native,
+        hypERC20contracts[TokenType.native],
+        [config.mailbox],
+      );
     }
   }
 
   protected async deploySynthetic(
     chain: ChainName,
     config: HypERC20Config,
   ): Promise<HypERC20> {
-    const router: HypERC20 = await this.deployContract(
+    const tokenType = isFastConfig(config)
+      ? TokenType.fastSynthetic
+      : TokenType.synthetic;
+    const router: HypERC20 = await this.deployContractWithName(
       chain,
-      isFastConfig(config) ? TokenType.fastSynthetic : TokenType.synthetic,
+      tokenType,
+      hypERC20contracts[tokenType],
```

Comment:
ditto here for config.type ?

Reply:
in this particular case, `config.type` could be fastsynthetic/synthetic/syntheticuri, and we have to narrow that down to one of fastsynthetic/synthetic